pkg/controller/certificates/sync.go

```diff
@@ -407,13 +411,25 @@ func (c *certificateRequestManager) processCertificate(ctx context.Context, crt
 	}
 
 	reason := apiutil.CertificateRequestReadyReason(existingReq)
+
+	// If the CertificateRequest condition is present and has the status of
+	// "True" then do not attempt to retry the CertificateRequest. Else we can
+	// retry.
+	if apiutil.CertificateRequestHasInvalidRequest(existingReq) {
+		log.Info("CertificateRequest is in an InvalidRequest state and will no longer be processed", "state", reason)
+
+		c.recorder.Eventf(crt, corev1.EventTypeWarning, "CertificateRequestInvalidRequest", "The failed CertificateRequest %q is an invalid request and will no longer be processed", existingReq.Name)
```

Comment:
Update this to include the reason and message and be more centred around resolving the issue IMO - given that a user requested a particular Certificate spec, it's fair to surface the error about the spec on the Certificate resource.

Also, we should probably report this on the Certificate's `status` somewhere too, and not just as an Event. I'm not certain _where_ we are best to do that though, given the 'Ready' condition solely maps to the readiness of the x509 certificate stored in the Secret (compared to the certificate that the user is requesting). It'd be misleading to report a certificate as 'not ready' if an existing, valid and up to date certificate exists and requesting the *new* one is failing.

Reply:
:+1: